lib/proxy/http_download.rb

```diff
@@ -14,11 +14,15 @@ def initialize(src, dst, read_timeout = nil, connect_timeout = nil, dns_timeout
       dns_timeout ||= DEFAULT_CONNECT_TIMEOUT
       connect_timeout ||= DEFAULT_DNS_TIMEOUT
 
-      args = [wget, "--connect-timeout=#{connect_timeout}",
+      args = ["--connect-timeout=#{connect_timeout}",
               "--dns-timeout=#{dns_timeout}",
               "--read-timeout=#{read_timeout}",
-              "--tries=3", "-nv", "-c", src.to_s, "-O", dst.to_s]
-      args << "--no-check-certificate" unless verify_server_cert
+              "--tries=3",
+              "--timestamping", # turn on timestamping to prevent redownloads
+              "--no-if-modified-since", # but use HTTP HEAD instead If-Modified-Since
```

Comment:
--no-if-modified-since doesn't work for EL7 wget. This will break it for proxies running on EL7.

Reply:
Damn, I did not expect a recent argument in wget and tested on Fedora. Does it work reliably with Katello without this option?

I need to test this on RHEL7 with Katello if this is needed.

Reply:
> The argument is useful, some mirrors do not implement the header. Maybe a setting to be able to enable this? But on RHEL7 this will redownload the file from mirrors which do not support this.

I am afraid timestamping won't be helpful at all. Manual page of wget says:
```
For this reason, -N (for timestamp-checking) is not supported in
combination with -O: since file is always newly created, it will
always have a very new timestamp. A warning will be issued if this
combination is used.
```

Trying it warns about it:
```
$ wget -N -O /dev/null
WARNING: timestamping does nothing in combination with -O. See the manual
for details.
```

Reply:
We don't use `-O /dev/null`, where did you find that?